### PR TITLE
Change "fail" to "stopAndFail" 🧐

### PR DESCRIPTION
**Pull request by bobcatfish**
# Changes

This commit proposes changes the wording of the `fail` option for
`onError` from `fail` to make it more explicitly clear that this option
(which is the default behavior when `onError` is not specified) will
both cause the Task to fail AND stop execution of any subsequent steps.

This change doesn't propose changing any of the described functionality,
just the value of the field.

Change to the TEP should be approved before this is merged
(https://github.com/tektoncd/community/pull/497) and if we go ahead with
it we should do a patch release ASAP to minimize user impact - that
being said, this change impacts the default value of the field which
users are unlikely to specify explicitly - users are much more likely to
start using `onError: continue` and for them there would be no impact.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
⚠️ Backwards incompatible change to new onError feature ⚠️ the value `fail` has been renamed to `stopAndFail` - if you are not using the `onError` field or you are using `onError: continue` you will not be impacted.
```

### CODE REVIEW

**Comment by bobcatfish** [comment]
Since this is technically a backwards incompatible change, [we need approval from > 50% of owners](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#backwards-incompatible-changes). PTAL @tektoncd/core-maintainers

**Review by sbwsg — APPROVED** [review]

**Review by dibyom — APPROVED** [review]
This definitely sounds clearer!

**Review by pritidesai — APPROVED** [review]
sounds good, thanks @bobcatfish

**Comment by bobcatfish** [comment]
alternatively we could always call it:

image

😇

**Review by imjasonh — APPROVED** [review]

**Comment by bobcatfish** [comment]
Thanks all!!

/hold cancel

**Pull request merged**
